app/metal-controller-manager/api/v1alpha1/serverclass_types.go

```diff
@@ -22,6 +22,7 @@ type Qualifiers struct {
 type ServerClassSpec struct {
 	EnvironmentRef *corev1.ObjectReference `json:"environmentRef,omitempty"`
 	Qualifiers     Qualifiers              `json:"qualifiers"`
+	Selector       *metav1.LabelSelector   `json:"selector"`
```

Comment:
It seems that `metav1.LabelSelector` is typically used as a value field, not as a pointer field. `metav1.LabelSelectorAsSelector` will convert empty non-nil LabelSelector to `labels.Everything()` which is what we need, right? That should simplify some logic.

Reply:
Good point. Fixed.